### ISSUE TITLE
CPUParticles2D One Shot while emitting bug

### ISSUE THREAD

**Issue by mani144** [godotengine/godot#24848]
**Godot version:** Godot 3.1 Beta 1

**OS/device including version:** Windows 10 , 2080ti

**Issue description:** it`s small bug regarding Emitting and one shot for CPUParticles2D, the problem is when the particles is emitting you can use OneShot only once, i dont know how can you improve it however i think the behavior should be like this : 
if Emitting is running then you can use One Shot by checking it and unchecking it the Emitter should reset every time one shot is checked again ( Suggestion or Bug ?)

**Steps to reproduce:**
![20190108200552](https://user-images.githubusercontent.com/17520497/50847646-2ee65000-1383-11e9-9786-18a01fc2217a.gif)

**Minimal reproduction project:**

**Comment by qarmin** [comment]
Duplicate #21794

**Comment by mani144** [comment]
I thought cpu particles have different structure that normal particles. If that is the case then let me know so that I close it.

**Comment by Chaosus** [comment]
Closed as duplicate of #21794

**Comment by Chaosus** [comment]
> I thought cpu particles have different structure that normal particles. 

I've added note about CPUParticles2D to the parent issue
